tools/generator/filters/java.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2020 Sacha Delanoue
+# Copyright (c) 2020 Antoine Pietri
+
+from functools import partial
+
+from . import register_filter
+from .common import camel_case, generic_args, generic_prototype, is_array
+from .cxx import cxx_comment, cxx_type
+
+
+@register_filter
+def java_type(value: str) -> str:
+    if value in ('byte', 'char', 'short', 'int', 'long', 'float', 'double', 'void'):
+        return value
+    if value == 'bool':
+        return "boolean"
+    if value == "string":
+        return "String"
+    if value.endswith(" array"):
+        return "{}[]".format(java_type(value[:-len(' array')]))
+    return camel_case(value)
+
+
+@register_filter
+def java_args(value, type_mapper=lambda x: x) -> str:
+    return generic_args(value, type_mapper=java_type)
+
+
+java_prototype = register_filter(
+    partial(generic_prototype, type_mapper=java_type, arg_mapper=java_args),
+    name='java_prototype',
+)
+
+java_comment = register_filter(cxx_comment, name='java_comment')
+
+
+def _jtype(typename: str) -> str:
+    """Return the type to be used as Lang in lang2cxx and cxx2lang"""
+    generics = {
+        'bool': 'jboolean',
+        'double': 'jdouble',
+        'int': 'jint',
+        'string': 'jstring',
+    }
+    if typename in generics.keys():
+        return generics[typename]
+    return camel_case(typename)
+
+
+def _cxx_type(value: str) -> str:
+    if value == "bool":
+        return "boolean"
+    return cxx_type(value)
+
+
+@register_filter
+def java_to_cxx(value: str) -> str:
+    if is_array(value):
+        base_type = value[:-len(' array')]
+        return 'lang2cxx_array<{}, {}>'.format(
+            _jtype(base_type),
+            _cxx_type(base_type)
+        )
+    elif value in ('int', 'double', 'bool', 'string'):
+        return 'lang2cxx<{}, {}>'.format(_jtype(value), _cxx_type(value))
+    else:
+        return 'lang2cxx<jobject, {}>'.format(_cxx_type(value))
+
+
+@register_filter
+def cxx_to_java(value: str) -> str:
+    if is_array(value):
+        base_type = value[:-len(' array')]
+        return 'cxx2lang_array<{}, {}>'.format(
+            _cxx_type(base_type),
+            _jtype(base_type)
+        )
+    elif value in ('int', 'double', 'bool', 'string'):
+        return 'cxx2lang<{}, {}>'.format(_cxx_type(value), _jtype(value))
+    else:
+        return 'cxx2lang<{}, jobject>'.format(_cxx_type(value))
+
+
+@register_filter
+def get_signature(value: str) -> str:
+    if is_array(value):
+        return "[" + get_signature(value[:-len(' array')])
+    dic = {
+        'bool': 'Z',
+        'double': 'D',
+        'int': 'I',
+        'void': 'V',
+        'string': 'Ljava/lang/String;',
+    }
+    if value in dic.keys():
+        return dic[value]
+    return 'L' + camel_case(value) + ';'
+
+
+@register_filter
+def get_fn_signature(func) -> str:
+    return "(" + "".join(get_signature(arg_type) for _, arg_type, _ in func['fct_arg']) + ")" + get_signature(func['fct_ret_type'])
```

Comment:
PEP8

Please add docstrings to explain the difference between a signature and a prototype?

Reply:
I renamed the function and added a comment.